src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -35,7 +35,7 @@ public async Task<GeneratedCodeWorkspace> ExecuteAsync(Task<CodeModel> codeModel
             ValidateConfiguration (configuration);
 
             Directory.CreateDirectory(configuration.OutputFolder);
-            var projectDirectory = Path.Combine(configuration.OutputFolder, Configuration.ProjectRelativeDirectory);
+            var projectDirectory = Path.Combine(configuration.OutputFolder, configuration.ProjectRelativeDirectory!);
```

Comment:
I wonder if we should just call it `ProjectDirectory` and add somewhere a description that if it relative, it will be resolved from `OutputFolder`. `Path.Combine` will ignore the first argument, so absolute path should just work.

Reply:
Good suggestion. There is another place I'm not sure whether absolute path works and I will verify it:

https://github.com/Azure/autorest.csharp/blob/5e9f905be1deb76ece2ef6a721d8b09cf1b65501/src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpProj.cs#L122

I also discovered that autorest already has a [shared flag](https://github.com/Azure/autorest/blob/main/docs/generate/flags.md#shared-flags) called `project-folder`, so we can reuse that.

Reply:
Verified that we have to use relative path for the above code snippet.